boa_temporal/src/fields.rs

```diff
@@ -0,0 +1,487 @@
+                    result.push(("minute".to_owned(), FieldValue::Integer(self.minute)));
+                }
+                FieldMap::SECOND => {
+                    result.push(("second".to_owned(), FieldValue::Integer(self.second)));
+                }
+                FieldMap::MILLISECOND => result.push((
+                    "millisecond".to_owned(),
+                    FieldValue::Integer(self.millisecond),
+                )),
+                FieldMap::MICROSECOND => result.push((
+                    "microsecond".to_owned(),
+                    FieldValue::Integer(self.microsecond),
+                )),
+                FieldMap::NANOSECOND => result.push((
+                    "nanosecond".to_owned(),
+                    FieldValue::Integer(self.nanosecond),
+                )),
+                FieldMap::OFFSET => result.push((
+                    "offset".to_owned(),
+                    self.offset
+                        .clone()
+                        .map_or(FieldValue::Undefined, FieldValue::String),
+                )),
+                FieldMap::ERA => result.push((
+                    "era".to_owned(),
+                    self.era
+                        .map_or(FieldValue::Undefined, |s| FieldValue::String(s.to_string())),
+                )),
+                FieldMap::ERA_YEAR => result.push((
+                    "eraYear".to_owned(),
+                    self.era_year
+                        .map_or(FieldValue::Undefined, FieldValue::Integer),
+                )),
+                FieldMap::TIME_ZONE => result.push((
+                    "timeZone".to_owned(),
+                    self.time_zone
+                        .clone()
+                        .map_or(FieldValue::Undefined, FieldValue::String),
+                )),
+                _ => {}
+            }
+        }
+
+        result
+    }
+
+    /// Resolve `TemporalFields` month and monthCode fields.
+    pub(crate) fn iso_resolve_month(&mut self) -> TemporalResult<()> {
+        if self.month_code.is_none() {
+            if self.month.is_some() {
+                return Ok(());
+            }
+
+            return Err(TemporalError::range()
+                .with_message("month and MonthCode values cannot both be undefined."));
+        }
+
+        let unresolved_month_code = self
+            .month_code
+            .as_ref()
+            .expect("monthCode must exist at this point.");
+
+        let month_code_integer = month_code_to_integer(*unresolved_month_code)?;
+
+        let new_month = match self.month {
+            Some(month) if month != month_code_integer => {
+                return Err(
+                    TemporalError::range().with_message("month and monthCode cannot be resolved.")
+                )
+            }
+            _ => month_code_integer,
+        };
+
+        self.month = Some(new_month);
+
+        Ok(())
+    }
+}
+
+fn month_code_to_integer(mc: TinyStr4) -> TemporalResult<i32> {
```

Comment:
Could probably use `TinyAsciiStr<4>` instead, since this type has an ominous warning:

`These are temporary compatability reexports that will be removed in a future version.`